### ISSUE TITLE
NVDA can't read the expanded content of a Primereact accordion with the Arrow-Down key

### ISSUE THREAD

**Issue by sylven32** [nvaccess/nvda#11250]
Steps:
1. install NVDA, start it and open the NVDA Speech Viewer
2. browse to https://www.primefaces.org/primereact/showcase/#/accordion
3. expand Godfather1 (under the Default title, same happens with Multiple)
4. press Arrow-Down, once or several times

Actual:
Nothing happens, except that the page is scrolling down (standard feature of arrow key)

Expected:
NVDA should be able to read the summary of the movie

The noticeable workaround is to Shift-Tab after step 3 then Arrow-Down several times up to the movie's summary.

It's unclear whether NVDA is failing to read correct markup or if Primereact Accordion is setting incorrect aria attributes. Another defect is open against Primereact (https://github.com/primefaces/primereact/issues/1406) for this project's team to analyze too.

#### NVDA installed/portable/running from source:
https://www.nvaccess.org/files/nvda/releases/2020.1/nvda_2020.1.exe
#### NVDA version:
2020.1
#### Windows version:
Windows 10
#### Name and version of other software in use when reproducing the issue:
Firefox 76.0.1
#### Does the issue still occur after restarting your computer?
Not tried, happens on another laptop of a colleague of mines
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No
#### If addons are disabled, is your problem still occuring?
Not tried
#### Did you try to run the COM registry fixing tool in NVDA menu / tools?
No

**Comment by DrSooom** [comment]
I can reproduce this with Firefox 77.0.1 and NVDA 2019.2.1 as long as the Speech Viewer is opened. Otherwise everything works as expected.

btw: Is it possible to use the html5 `<details><summary>` pair for such accordions instead of the method, which is provided by the above mentioned website?

**Comment by sylven32** [comment]
Many thanks for your quick answer. Sharing this information locally so that your proposed workaround is checked.
The mentioned website is the gallery of the React components library we're using in the software we're developing. We do not have any control on the markup they're generating, hence your details/summary proposal is not an option for us, while might make sense to them. They've been provided a link to this issue as well.

**Comment by sylven32** [comment]
We've checked the workaround (Speech View closed) with NVDA 2020.1 and did not succeed unfortunately.
Tried Chrome Version 83.0.4103.97 (Official Build) (64-bit) in addition to Firefox 76.0.1

**Comment by DrSooom** [comment]
@sylven32: Okay, try the following:
1. Move the Firefox/Chrome window to the left half of the screen and the Speech Viewer window to the right half. Does this help?
2. Or press CTRL+Enter instead of Enter alone on such tabs – even with enabled Speech Viewer. CTRL+Enter works here on my machine with enabled Speech Viewer – Enter alone not.
3. And for Firefox: Open "about:support" and search for "Accessibility". The values here must be: true, 0, true and NVDA. If not, close Firefox, restart NVDA and start Firefox again. Let us know if this had changed anything regarding these parameters.

**Comment by DrSooom** [comment]
Additional question: Are you using the Focus or the Browse Mode? You can toggle them by pressing NVDA+Space (= INS/NumINS/CapsLock+space). In Focus Mode it's normal that NVDA is not able to read "plain/protected" text like headlines, tables and paragraphs with the arrow keys. See chapter 6 – Browse Mode in the NVDA User Guide for more information.

All my tests so far were performed in Browse Mode. But even when I'm using the Focus Mode and press Enter on such a tab, NVDA correctly reports its state: expanded or collapsed.

And then U+E901 and U+E902 are also reported via Speech and Braille output by NVDA when you're in Focus Mode, but not in Browse Mode. Perhaps [aria-label](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-label_attribute) could help here to fix this as these two Unicode characters cannot be spoken by most TTS synthesizers by default. (This is just a tip and has nothing to do with the main issue here. I just recognized it during testing.)

**Comment by DrSooom** [comment]
@sylven32: Could you also please test this with [NVDA Alpha-20317](https://ci.appveyor.com/api/buildjobs/rbo0m2q24s33nd6x/artifacts/output/nvda_snapshot_alpha-20317,6bf3ae5f.exe) and report the results as it contains several Browse Mode bugfixes. Thanks.

**Comment by sylven32** [comment]
About the tries you've proposed (3 comments above), none of 1, 2 neither 3 did help.
About toggling between focus and browse mode, added a new shortcut (Ctrl-0) from the prefs as proposed ones didn't work on my laptop, then used that Ctrl-0 after the section is open with Space and Arrow Down -> this worked with 2020.1
About the alpha setup, exact same behaviors observed (no change compared to 2020.1)

I think browse mode setup is the best workaround here. I'll check internally. Many thanks for your great and quick help.

**Comment by Adriani90** [comment]
This works as expected in NVDA Version: alpha-20339,975d39d1, Firefox 77 and Chrome 83.
It is expected to work only in browse mode. The tab can be expanded both in focus and browse mode, but the description is readable only in browse mode since it is plain text and it cannot be focused via tabindex. So it is rather part of the virtual buffer and can be navigated only in browse mode. This is expected behavior and is consistent among all screen readers in Windows.
So for this to work correctly, do following:
1. Open NVDA
2. Open the website in the description and make sure NVDA is in browse mode
3. Navigate with h to the heading "default" or "multiple"
4. Press down arrow to land on "tab godfather 1"
5. Press enter
6. Press down arrow
Actual behavior: in step 5 you will hear "expanded"
in step 6 you will hear the first line of the description text.

This is how most blind users navigate such areas on websites.

I am closing this issue as works for me. But please feel free to comment on this if you have other findings and we can reopen the discussion. Thanks for reporting.

**Comment by DrSooom** [comment]
@Adriani90: Did you also test this with an opened Speech Viewer?

**Comment by Adriani90** [comment]
@DrSooom yes, it works still as expected.
